BitstreamStorageServiceImplIT tests fail on macOS

## Describe the bug

The tests in "dspace-api/src/test/java/org/dspace/storage/bitstore/BitstreamStorageServiceImplIT.java" fail when run on macOS:

```
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 18.94 s <<< FAILURE! -- in org.dspace.storage.bitstore.BitstreamStorageServiceImplIT
[ERROR] org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.testBatchCommitSizeThree -- Time elapsed: 0.598 s <<< ERROR!
java.lang.RuntimeException: java.io.IOException: java.io.IOException: Illegal bitstream path constructed
	at org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.createBitstream(BitstreamStorageServiceImplIT.java:205)
	at org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.createBitstreams(BitstreamStorageServiceImplIT.java:193)
...
Caused by: java.io.IOException: java.io.IOException: Illegal bitstream path constructed
	at org.dspace.storage.bitstore.DSBitStoreService.put(DSBitStoreService.java:126)
	at org.dspace.storage.bitstore.BitstreamStorageServiceImpl.store(BitstreamStorageServiceImpl.java:123)
	at org.dspace.content.BitstreamServiceImpl.create(BitstreamServiceImpl.java:124)
...
[ERROR] org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.testDefaultBatchCommitSize -- Time elapsed: 0.428 s <<< ERROR!
java.lang.RuntimeException: java.io.IOException: java.io.IOException: Illegal bitstream path constructed
	at org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.createBitstream(BitstreamStorageServiceImplIT.java:205)
	at org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.createBitstreams(BitstreamStorageServiceImplIT.java:193)
	at org.dspace.storage.bitstore.BitstreamStorageServiceImplIT.testDefaultBatchCommitSize(BitstreamStorageServiceImplIT.java:98)
...
Caused by: java.io.IOException: java.io.IOException: Illegal bitstream path constructed
	at org.dspace.storage.bitstore.DSBitStoreService.put(DSBitStoreService.java:126)
	at org.dspace.storage.bitstore.BitstreamStorageServiceImpl.store(BitstreamStorageServiceImpl.java:123)
	at org.dspace.content.BitstreamServiceImpl.create(BitstreamServiceImpl.java:124)
...
```

This error currently (September 26, 2025) only occurs on the "main" branch because the "BitstreamStorageServiceImplIT" integration tests have not yet been incorporated into a tagged release.

## To Reproduce

This issue likely only occurs on macOS, due to the way macOS handles the temporary directories in Java.

1) Clone the DSpace repository and switch into the directory:

```
$ git clone https://github.com/DSpace/DSpace.git
$ cd DSpace
```

2) Build DSpace:

```
$ mvn clean install
```

3) Run the "BitstreamStorageServiceImplIT" integration test:

```
$ cd dspace-api
$ mvn verify -Dit.test=BitstreamStorageServiceImplIT -DskipIntegrationTests=false
```

The tests will fail as indicated above.

## Expected behavior

That the "BitstreamStorageServiceImplIT" integration tests will pass successfully on macOs.

## Error Explanation

The "BitstreamStorageServiceImplIT" tests verify that bitstreams are properly stored in the asset store, even if the batch process that is adding the bitstreams is interrupted.

The "BitstreamStorageServiceImplIT" tests use a special test bitstore service, "LimitedTempDSBitStoreService", which relies on temporary folders (created by the "org.junit.rules.TemporaryFolder" rule) to ensure that the bitstream files are cleaned up when the tests are completed.

On macOs, the jUnit "TemporaryFolder" rule creates files within the "/var" directory. However, the "/var" directory is really just a symbolic link to the "/private/var" directory.

When the tests are running, the "Illegal bitstream path constructed" error is occurring because in "dspace-api/src/main/java/org/dspace/storage/bitstore/DSBitStoreService.java", there is the following [check](https://github.com/DSpace/DSpace/blob/4efcd69d9575b8745d599032602f2b33c100c573/dspace-api/src/main/java/org/dspace/storage/bitstore/DSBitStoreService.java#L259C9-L259C66):

```
if (!normalizedPath.startsWith(baseDir.getAbsolutePath())
```

On macOS, the "normalizedPath" will start with "/private/var" and the "baseDir.getAbsolutePath()" starts with "/var", failing the check and triggering the IOException. This check was added in https://github.com/DSpace/DSpace/commit/6799660a903bfea985c818654cca3891527780de and modified in https://github.com/DSpace/DSpace/commit/6c3274630ce081b8395168149756569689153ce0

Earlier in the code, the "normalizedPath" is generated by calling `baseDir.getCanonicalFile()`. The "getCanonicalFile()" method resolves symbolic links, which is why the "normalizedPath" starts with "/private/var". In the check above, however, the "baseDir.getAbsolutePath()" is used, and the "getAbsolutePath()" does _not_ resolve symbolic links, leading to the error.

This error does not occur in Linux because "/var" is not typically a symbolic link on Linux.

## Related work

* https://github.com/DSpace/DSpace/commit/6799660a903bfea985c818654cca3891527780de
* https://github.com/DSpace/DSpace/commit/6c3274630ce081b8395168149756569689153ce0

--- a/dspace-api/src/main/java/org/dspace/storage/bitstore/DSBitStoreService.java
+++ b/dspace-api/src/main/java/org/dspace/storage/bitstore/DSBitStoreService.java
@@ -256,11 +256,11 @@ public class DSBitStoreService extends BaseBitStoreService {
         Path normalizedPath = bitstreamFile.toPath().normalize();
         String[] allowedAssetstoreRoots = DSpaceServicesFactory.getInstance().getConfigurationService()
                 .getArrayProperty("assetstore.allowed.roots", new String[]{});
-        if (!normalizedPath.startsWith(baseDir.getAbsolutePath())
+        if (!normalizedPath.startsWith(baseDir.getCanonicalPath())
             && !StringUtils.startsWithAny(normalizedPath.toString(), allowedAssetstoreRoots)) {
             log.error("Bitstream path outside of assetstore root requested:" +
                     "bitstream={}, path={}, assetstore={}",
-                    bitstream.getID(), normalizedPath, baseDir.getAbsolutePath());
+                    bitstream.getID(), normalizedPath, baseDir.getCanonicalPath());
             throw new IOException("Illegal bitstream path constructed");
         }
         return bitstreamFile;

